crates/bitwarden-auth/src/registration.rs

```diff
@@ -33,4 +47,129 @@ impl RegistrationClient {
         let api_client = &client.get_api_configurations().await.api_client;
         // Do API request here. It will be authenticated using the client's tokens.
     }
+
+    /// Initializes a new cryptographic state for a user and posts it to the server; enrolls in
+    /// admin password reset and finally enrolls the user to TDE unlock.
+    pub async fn post_keys_for_tde_registration(
+        &self,
+        org_id: String,
+        org_public_key: B64,
+        // Note: Ideally these would be set for the register client, however no such functionality
+        // exists at the moment
+        user_id: String,
+        device_id: String,
+        trust_device: bool,
```

Comment:
❓ 

For these ID parameters can we use the `OrganizationId` `UserId` etc. as the input types from `crates/bitwarden-core/src/ids.rs`? Or is there some limitation for those?